echo-api/src/main/java/com/netflix/spinnaker/echo/api/events/NotificationParameter.java

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright 2020 Armory, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.spinnaker.echo.api.events;
+
+import lombok.Data;
+import lombok.NoArgsConstructor;
+import lombok.NonNull;
+
+/** Definition of a notification parameter that a user can configure. */
+@Data
+@NoArgsConstructor
+public class NotificationParameter {
```

Comment:
I stole this from Orca's preconfigured job stage parameters. Maybe someday if the abstraction is right this model could live in Kork and be shared across services that want Deck to generate form fields.